docs/crocks/Arrow.md

```diff
@@ -0,0 +1,77 @@
+# Arrow
+
+`Arrow a b`
+
+--
+
+--
+
+```js
+--
+```
+
+`Arrow` exposes the following functions on the constructor and instance:
+
+| Constructor | Instance |
+|:---|:---|
+| [`empty`](#empty), [`type`](#type) | [`both`](#both), [`concat`](#concat), [`contramap`](#contramap), [`empty`](#empty), [`first`](#first), [`inspect`](#inspect), [`map`](#map), [`promap`](#promap), [`runWith`](#runWith), [`second`](#second), [`type`](#type), [`value`](#value) |
+
+## Constructor
+
+### empty
+
+`Arrow m => () -> m () b`
+
+### type
+
+`() -> String`
+
+## Instance
+
+### both
+
+`Arrow m, Pair p => m a b ~> () -> m (p a b) (p c d)`
+
+### concat
+
+`Arrow m => m a b ~> m a b -> m a b`
```

Comment:
This would be something like this as we are not constrained to an mono-morphisim here and it is composition under the hood:
`Arrow m => m a b ~> m b c -> m a c`

We will keep it for now, but now that I see this, this should not be `concat`. Will need to do something else.